### ISSUE TITLE
Forward deleting empty paragraph before a list (or other block types)

### ISSUE THREAD

**Issue by mtias** [WordPress/gutenberg#21558]
Leave an empty paragraph and write a list after it. Now focus on the empty paragraph and do FN+Backspace. The list is lost and the content becomes paragraph. I'd expect it to just remove the empty paragraph and leave the list alone.

**Comment by draganescu** [comment]
I tested and can confirm this (by which occasion I also learned about fn + backspace = forward delete).

**Comment by kathrynwp** [comment]
I'm still able to replicate this with:

WordPress 6.0.1
Gutenberg 13.8.0 active or inactive
No other plugins active

I'm a longtime Mac user but also never knew about fn + backspace before!

@mtias would you like to keep this issue open or close it, since there's been no movement on it in quite a long time?

**Comment by mtias** [comment]
Yes, I was hoping the new list refactor would fix it, but appear to still be an issue

**Comment by Mamaduka** [comment]
@mtias, the experience should be improved after #45075.

**Comment by bph** [comment]
I can still reproduce it with fn+backspace, the empty paragraph stays and the first list item is converted to a paragraph. However,  if I hit backspace while focused on the empty paragraph, it removes the empty paragraph and leaves the list intact. 

https://github.com/WordPress/gutenberg/assets/39980/a14d4ce0-9cdc-484c-9361-3ddbbd654209

**Comment by ellatrix** [comment]
The current behaviour looks correct. Forward delete is equivalent to ArrowRight + Backspace, so that's what's happening. Also, the behaviour shouldn't change on wether or not the paragraph is empty I think. What should happen what the paragraph is not empty and selection at then end of it?

**Comment by draganescu** [comment]
I agree with @ellatrix that the behaviour is correct.

**Comment by mtias** [comment]
I disagree. Testing in Apple notes, forward delete is not equivalent to arrow right + backspace. If you have the following:

```
... caret is here on empty line ...
- One
- Two
```

If you do arrow right the cursor is placed at the beginning of the first list item; if you then press backspace it unindents the list. This is fine.

But if you press Fn+Backspace it doesn't do the same — it collapses the empty line and preserves the list. I'd expect the same here.

**Comment by draganescu** [comment]
Is this the right behavior though? It's supposed to delete chars to the right (or opposite to backspace depending on lang dir)

When cursor is on empty new line it is _after_ the invisible new line char, so forward delete removes the marker that is next, and next is the list marker. 


Wikipedia says:

> ... forward delete, discards the character ahead of the cursor's position, moving all following characters one position "back" towards the freed letterspace. 

Maybe the point of the ticket is not to fix but to enhance the behavior to emulate standard text editing _conventions_?

**Comment by ellatrix** [comment]
Alternatively we immediately transform the next list item to a paragraph AND delete the paragraph, which is what happens now after pressing Delete twice. I thought it's nice to have an intermediate step but maybe it's weird.

For Backspace that would also mean immediately merge a list item into a paragraph instead of first "removing" the bullet.
